observed.py

```diff
@@ -87,7 +88,8 @@ class ObserverFunction:
     does not prevent garbage collection of the observing function.
     """
 
-    def __init__(self, func, identify_observed, weakref_info):
+    def __init__(self, func: Callable, identify_observed: bool,
```

Comment:
Is it possible to use a more specific type here, e.g. `Callable[T, R]` or something like that?